gtk4/sample/examples/file_dialog_demo.rb

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2024  Ruby-GNOME Project Team
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+# Original:
+#   https://gitlab.gnome.org/GNOME/gtk/-/blob/main/demos/gtk-demo/main.c
+#
+# See the following for license information:
+#   * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/AUTHORS
+#   * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/COPYING
+
+# We need to use "GtkDemo" as type name because "menu-listitem.ui"
+# uses "GtkDemo" as type name.
```

Comment:
Do we need this comment?

Reply:
No.  Delete it.

My main objection to this example is that it uses the begin/rescue/send to capture when the user presses the cancel button.  It seems super-hacky to me.  Im sure there's a correct solution, but I can't find it.